### DISCUSSION TITLE
Feature: Extend Logs and AccessRoles

### DISCUSSION THREAD

**Discussion by bloor** [pocketbase/pocketbase#474] [Q&A] [answered]
1) We already have "created" and "updated" by default on each table, which is nice. But in some instances "created_by" and "updated_by"
 are good to have.

 
2) Add the "user" in API request events/logs.
 
3) Currently there is only 1 tier of admins (superadmins) so you can;t create "Editor" accounts that can view/manipulate the database in restricted mode. Because of this I think it limits the potential as a basic CMS. As such, for my project I will still have to build a separate admin-dashboard and use the poketbase UI just as a glorified phpmyadmin.

**Comment by ganigeorgiev** [comment, marked as answer]
1. Default `created_by` and `updated_by` will complicate how users will be managed after #376, since we will support multiple auth collections and it is better to leave to the developers to manually create the fields if they need them.

2. I'll consider logging the user id and collection, but that will happen after #376.

3. This was requested several times, but at least for v1.0.0 I'm not planning introducing admin roles since it will complicate how the access checks are performed (we will end up having guests, multiple admin roles and multiple auth collections constrained by the api rules at the same time and I think it will be confusing for the end users).

**Comment by osseonews** [comment]
I'd upvote suggestion 3 above, having access restrictions for other types of admins, and I think it's critical for expanded use of Pocketbase. After experimenting with Pocketbase alot today, I'd say your UI for the Admin is simply amazing. It would be a total waste of time for users to have to build their own separate admin-dashboard to accommodate additional admin "types" so they can restrict access to certain features in the admin dashboard UI for certain admins. There should be as suggested different admin roles, like "Editor", and you should be able to restrict access based on the role. I guess this is somewhat similar to how access is restricted right now for  users in the API rules for collections.
